edge/pkg/metamanager/metaserver/handlerfactory/handler.go

```diff
@@ -49,6 +49,38 @@ func NewFactory() *Factory {
 	return &f
 }
 
+func (f *Factory) Version() http.Handler {
+	if h, ok := f.getHandler("version"); ok {
+		return h
+	}
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	h := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		result := []byte("{\"major\":\"1\", \"minor\":\"22\", \"gitVersion\":\"v1.22.17\", \"gitCommit\":\"a7736eaf34d823d7652415337ac0ad06db9167fc\", \"gitTreeState\":\"clean\", \"buildDate\":\"2022-12-08T11:42:04Z\", \"goVersion\":\"go1.16.15\", \"compiler\":\"gc\", \"platform\":\"linux/amd64\"}")
```

Comment:
the question is should we always reply k8s version against this request, right? that said, when starting metaserver, it should proxy the request to the api-server to ask this information at least once? after all, this is just a temporary patch to return the static version as it is used for our test.

Reply:
okay, lets say this is open issue with https://github.com/kubeedge/kubeedge/issues/4844#issuecomment-1653498983

Reply:
https://github.com/kubeedge/kubeedge/pull/4959

Reply:
okay, i see the update. resolving this comment.